ctru-rs/src/gfx.rs

```diff
@@ -9,17 +10,40 @@ use crate::services::gspgpu::{self, FramebufferFormat};
 /// provides helper functions and utilities for software rendering.
 ///
 /// The service exits when this struct is dropped.
-pub struct Gfx(());
+pub struct Gfx {
+    top_screen: RefCell<TopScreen>,
+    bottom_screen: RefCell<BottomScreen>,
```

Comment:
I recommend making these fields public like in the diff I shared. There's not much reason to hide them (since we can use something like `_private: ()` to keep the struct init private), and it saves a lot of code duplication (you're just re-declaring the borrow methods from Rc).